manual/manual_zx.rst

```diff
@@ -0,0 +1,654 @@
+
+The :py:class:`Flow` object that is returned abstracts away the partial ordering of the measured qubits of the diagram by just giving the depth from the outputs, i.e. all output qubits and those with no corrections have depth :math:`0`, all qubits with depth :math:`n` can be measured simultaneously and only require corrections on qubits at depth strictly less than :math:`n`. The measurement corrections can also be inferred from the flow, where :py:meth:`Flow.c()` gives the correction set for a given measured qubit (the qubits which require an :math:`X` correction if a measurement error occurs) and :py:meth:`Flow.odd()` gives its odd neighbourhood (the qubits which require a :math:`Z` correction).
+
+.. jupyter-execute::
+
+    from pytket.zx import Flow
+
+    fl = Flow.identify_pauli_flow(diag)
+
+    # We can look up the flow data for a particular vertex
+    # For example, let's take the first input qubit
+    vertex_ids = { v : i for (i, v) in enumerate(diag.vertices) }
+    in0 = diag.get_boundary(ZXType.Input)[0]
+    v = diag.neighbours(in0)[0]
+    print(vertex_ids[v])
+    print(fl.d(v))
+    print([vertex_ids[c] for c in fl.c(v)])
+    print([vertex_ids[o] for o in fl.odd(v, diag)])
+
+    # Or we can obtain the entire flow as maps for easy iteration
+    print({ vertex_ids[v] : d for (v, d) in fl.dmap.items() })
+    print({ vertex_ids[v] : [vertex_ids[c] for c in cs] for (v, cs) in fl.cmap.items() })
+
+.. note:: In accordance with the Pauli flow criteria, :py:meth:`Flow.c()` and :py:meth:`Flow.odd()` may return qubits that have already been measured, but this may only happen in cases where the required correction would not have affected the past measurement such as a :math:`Z` on a :py:class:`ZXType.PZ` qubit.
+
+.. Verification and focussing
+
+In general, multiple valid flows may exist for a given diagram, but a pattern with equal numbers of inputs and outputs will always have a unique focussed flow (where the corrections permitted on each qubit are restricted to be a single Pauli based on its label, e.g. if qubit :math:`q` is labelled as :py:class:`ZXType.XY`, then we may only apply :math:`X` corrections to :math:`q`). Given any flow, we may transform it to a focussed flow using :py:meth:`Flow.focus()`.
+
+.. Warning that does not update on rewriting
+
+.. warning:: A :py:class:`Flow` object is always with respect to a particular :py:class:`ZXDiagram` in a particular state. It cannot be applied to other diagrams and does not automatically update on rewriting the diagram.
+
+Conversions & Extraction
```

Comment:
Just to clarify a couple things around conversion.

Which extraction method(s) do we have available in pytket at the moment? It wasn't obvious to me from reading this section.

Is it possible to say which properties a circuit must have to be converted efficently? And which operations cause problems with the conversion?

Reply:
> Which extraction method(s) do we have available in pytket at the moment? It wasn't obvious to me from reading this section.

The paragraph starting "The current method implemented..." links to the paper introducing the extraction method and clarifies that this is the only such method currently implemented in pytket.

> Is it possible to say which properties a circuit must have to be converted efficently? And which operations cause problems with the conversion?

It should work for any MBQC diagram (i.e. using only the MBQC generators with appropriate handling of boundaries) that has gflow and corresponds to a unitary. This is covered in the same paragraph and I have added a note about the requirement for MBQC generators in the final code sample